moveit_core/robot_model/src/planar_joint_model.cpp

```diff
@@ -146,11 +146,23 @@ void computeTurnDriveTurnGeometry(const double* from, const double* to, double&
 {
   dx = to[0] - from[0];
   dy = to[1] - from[1];
-  drive_angle = atan2(dy, dx);
-
-  initial_turn = angles::shortest_angular_distance(from[2], drive_angle);
+  const double angle_straight_diff =
+      std::hypot(dx, dy) > 0.01 ? angles::shortest_angular_distance(from[2], std::atan2(dy, dx)) : 0.0;
+  const double angle_backward_diff = angles::normalize_angle(angle_straight_diff - M_PI);
```

Comment:
Maybe add `using namespace boost::math::double_constants;` at the top so that you can use `pi` everywhere?
```suggestion
  const double angle_backward_diff = angles::normalize_angle(angle_straight_diff - boost::math::constants::pi<double>);
```